src/__tests__/card-number.test.tsx

```diff
@@ -40,6 +40,16 @@ describe('Card number', () => {
     fireEvent.change(inputEl, { target: { value: '54233320332990' } });
 
     const errorEl = screen.getByTestId('cardNumberErrorEl');
-    expect(errorEl.textContent).toBe('Card number is invalid');
+    expect(errorEl).toBeInTheDocument();
```

Comment:
This test pattern `toBeInDocument` might not work. That's because only the textContent of the selected Element is conditionally rendered not the element itself.